gax/operations_stub.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAPIC_GENERATOR_CPP_GAX_OPERATIONS_STUB_H_
+#define GAPIC_GENERATOR_CPP_GAX_OPERATIONS_STUB_H_
+
+#include "google/longrunning/operations.pb.h"
+
+#include "call_context.h"
+#include "status.h"
+
+namespace google {
+namespace gax {
+
+class OperationsStub {
+ public:
+  virtual ~OperationsStub() = 0;
+
+  virtual gax::Status GetOperation(
+      gax::CallContext& context,
+      google::longrunning::GetOperationRequest const& request,
+      google::longrunning::Operation* response);
+
+  virtual gax::Status DeleteOperation(
+      gax::CallContext& context,
+      google::longrunning::DeleteOperationRequest const& request,
+      google::protobuf::Empty* response);
+
+  virtual gax::Status CancelOperation(
+      gax::CallContext& context,
+      google::longrunning::CancelOperationRequest const& request,
+      google::protobuf::Empty* response);
+
+  static constexpr MethodInfo get_operation_info_ = {
```

Comment:
I think the style guide says that public member variables do not have a trailing `_`. You could also consider a static member function that returns a reference to this structure.

Reply:
The intended pattern is for client methods to construct a CallContext from the appropriate MethodInfo struct, and for users to interact with MethodInfo only as a return from CallContext's accessor.

Minor tangent: during the design of CallContext, one pattern that seemed cool was that, since MethodInfo is immutable data produced at compile time, to have CallContext only store info by reference. It's safe, but it's not obviously safe to the casual reader, and it felt like a little bit of a premature optimization.